storage/database/dynamodb.go

```diff
@@ -0,0 +1,477 @@
+				go dynamoDB.createBatchWriteWorker()
+			}
+			dynamoDB.logger.Info("made dynamo batch write workers", "workerNum", WorkerNum)
+			return dynamoDB, nil
+		case dynamodb.TableStatusDeleting, dynamodb.TableStatusArchiving, dynamodb.TableStatusArchived:
+			return nil, errors.New("failed to get DynamoDB table, table status : " + tableStatus)
+		default:
+			dynamoDB.logger.Info("waiting for the table to be ready", "table status", tableStatus)
+			time.Sleep(1 * time.Second)
+		}
+	}
+}
+
+func (dynamo *dynamoDB) createTable() error {
+	input := &dynamodb.CreateTableInput{
+		// TODO-Klaytn: enable Provisioned mode
+		BillingMode: aws.String("PAY_PER_REQUEST"),
+		AttributeDefinitions: []*dynamodb.AttributeDefinition{
+			{
+				AttributeName: aws.String("Key"),
+				AttributeType: aws.String("B"), // B - the attribute is of type Binary
+			},
+		},
+		KeySchema: []*dynamodb.KeySchemaElement{
+			{
+				AttributeName: aws.String("Key"),
+				KeyType:       aws.String("HASH"), // HASH - partition key, RANGE - sort key
+			},
+		},
+		// ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+		// 	dynamoReadCapacityUnits:  aws.Int64(dynamo.config.dynamoReadCapacityUnits),
+		// 	dynamoWriteCapacityUnits: aws.Int64(dynamo.config.dynamoWriteCapacityUnits),
+		// },
+		TableName: aws.String(dynamo.config.TableName),
+	}
+
+	_, err := dynamo.db.CreateTable(input)
+	if err != nil {
+		dynamo.logger.Error("Error while creating the DynamoDB table", "err", err, "tableName", dynamo.config.TableName)
+		return err
+	}
+	dynamo.logger.Info("Successfully created the Dynamo table", "tableName", dynamo.config.TableName)
+	return nil
+}
+
+func (dynamo *dynamoDB) deleteTable() error {
+	if _, err := dynamo.db.DeleteTable(&dynamodb.DeleteTableInput{TableName: &dynamo.config.TableName}); err != nil {
+		dynamo.logger.Error("Error while deleting the DynamoDB table", "tableName", dynamo.config.TableName)
+		return err
+	}
+	dynamo.logger.Info("Successfully deleted the DynamoDB table", "tableName", dynamo.config.TableName)
+	return nil
+}
+
+func (dynamo *dynamoDB) tableStatus() (string, error) {
+	desc, err := dynamo.tableDescription()
+	if err != nil {
+		return "", err
+	}
+
+	return *desc.TableStatus, nil
+}
+
+func (dynamo *dynamoDB) tableDescription() (*dynamodb.TableDescription, error) {
+	describe, err := dynamo.db.DescribeTable(&dynamodb.DescribeTableInput{TableName: aws.String(dynamo.config.TableName)})
+	if describe == nil {
+		return nil, err
+	}
+
+	return describe.Table, err
+}
+
+func (dynamo *dynamoDB) Type() DBType {
+	return DynamoDB
+}
+
+// Path returns the path to the database directory.
+func (dynamo *dynamoDB) Path() string {
+	return fmt.Sprintf("%s-%s", dynamo.config.Region, dynamo.config.Endpoint)
+}
+
+// Put inserts the given key and value pair to the database.
+func (dynamo *dynamoDB) Put(key []byte, val []byte) error {
+	if len(key) == 0 {
+		return nil
+	}
+
+	if len(key)+len(val) > dynamoWriteSizeLimit {
```

Comment:
```suggestion
	if len(marshaledData) > dynamoWriteSizeLimit {
```
I think the checking is better placed after the marshaling.

Reply:
Because `marshaledData` is a map, it needs to be iterated all items to calculate the length.

Reply:
My bad, you can go with your way :)